### ISSUE TITLE
[Feature Request] Allowing <Redirect /> in customRoutes

### ISSUE THREAD

**Issue by kopax** [marmelab/react-admin#2767]
Hi @marmelab people,

We are reworking we react-admin and we are happy to see all the improvment.

I've been trying to add a `<Redirect />` component and by looking at the code of `CoreAdmin.tsx` and `RoutesWithLayout.js`, it seems that the `<Redirect />` cannot be processed properly.

This feature request consists of allowing `<Redirect />` component in `customRoutes`.

If you are interested in such a feature, I can implement it.

So far, we handled redirection using `<Redirect />` in our react app, is there another strategy with react-admin?

Thanks!.

**Comment by fzaninotto** [comment]
Hi @kopax,

Can you explain your use case? I need to understand the user experience you want to build.

**Comment by kopax** [comment]
Hi @fzaninotto, my use case is when a page get moved to a new path, we'd like to add a redirection from the old path in case people save those in their bookmarks.

**Comment by fzaninotto** [comment]
OK, thanks. Can you explain how you tried to do the redirection and why it failed?

**Comment by kopax** [comment]
I have tried to add one case in `customRoutes`:

```js
<Route name="new" component={NewPage} path="/path-new" exact={true} />
<Redirect name="old" from="/path-old" to="/path-new" />
```

It failed because `<Redirect />` is not created.

- https://github.com/marmelab/react-admin/blob/v2.6.0/packages/ra-core/src/CoreAdminRouter.js#L139
- https://github.com/marmelab/react-admin/blob/v2.6.0/packages/ra-core/src/CoreAdminRouter.js#L156
- https://github.com/marmelab/react-admin/blob/v2.6.0/packages/ra-core/src/RoutesWithLayout.js#L18

, only the `<Route />`. Also, this break the `catchAll` route.

**Comment by fzaninotto** [comment]
Understood, thanks. I'm marking it as en enhancement. We won't work on it in the near future, so you're welcome to submit a pull request against the `next` branch to add it.

**Comment by fzaninotto** [comment]
Fixed by #2771
